modules/compiler/utils/source/work_item_loops_pass.cpp

```diff
@@ -1714,7 +1714,9 @@ Function *compiler::utils::WorkItemLoopsPass::makeWrapperFunction(
 
   // Remap any constant expression which take a reference to the old function
   // FIXME: What about the main function?
-  for (auto *user : refF.users()) {
+  for (auto it = refF.user_begin(); it != refF.user_end();) {
+    auto user = *it;
+    it++;
```

Comment:
For `user`, can we continue to have `auto *user` or does that not work? For the iterator, that should either be `user = *it++;` or `user = *it; ++it;`, but for non-builtin-types, best to use pre-increment when you don't use the result.

Reply:
No issues with `auto *`, and I've changed to use the pre increment operator, thank you